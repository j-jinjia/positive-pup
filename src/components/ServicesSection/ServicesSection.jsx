import Button from "../Button/Button";
import ServicesCard from "../ServicesCard/ServicesCard";
import "./ServicesSection.scss";

const ServicesSection = ({ link }) => {
  return (
    <div className="services-section">
      <h2 className="services-section__header">Services I Offer</h2>
      <p className="services-section__sub-header">
        A breakdown of all the different kind of services I offer.
      </p>
      <hr className="services-section__line-break"></hr>
      <div className="services-section__course-container">
        <ServicesCard
          title="Online Courses"
          description=" Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quia quod
          ipsum saepe ipsa optio dolorem veniam tenetur nemo labore ut"
        />
        <ServicesCard
          title="Group Courses"
          description=" Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quia quod
          ipsum saepe ipsa optio dolorem veniam tenetur nemo labore ut"
        />
        <ServicesCard
          title="Bespoke Packages"
          description=" Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quia quod
          ipsum saepe ipsa optio dolorem veniam tenetur nemo labore ut"
        />
      </div>
      <Button buttonText="VIEW OUR SERVICES" isSecondary={true} link={link} />
    </div>
  );
};

export default ServicesSection;
