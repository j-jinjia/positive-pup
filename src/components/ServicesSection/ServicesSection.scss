@use "../../styles/variables" as *;

.services-section {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 2rem;
  min-height: 100vh;
  justify-content: center;

  &__course-container {
    margin-bottom: 1rem;
  }

  &__header {
    font-size: $font-size-h2;
    margin-bottom: 1rem;
  }
  &__sub-header {
    text-align: center;
    margin-bottom: 1.5rem;
    color: $colour-tertiary;
    font-weight: 400;
  }
  &__line-break {
    border: 1px solid $colour-primary;
    background-color: $colour-primary;
    width: 3.75rem;
    margin-bottom: 1.5rem;
  }
}

@media only screen and (min-width: $desktop-breakpoint) {
  .services-section {
    &__line-break {
      margin-bottom: 3rem;
    }
    &__course-container {
      display: flex;
      gap: 1rem;
    }
  }
}
