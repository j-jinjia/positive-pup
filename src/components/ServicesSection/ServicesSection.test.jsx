import { customRender } from "../../utils/testUtils";
import ServicesSection from "./ServicesSection";

it("should match ServicesSection component snapshot", () => {
  const { container } = customRender(<ServicesSection />);

  expect(container).toMatchSnapshot();
});
