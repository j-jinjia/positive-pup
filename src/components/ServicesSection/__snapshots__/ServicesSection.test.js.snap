// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match ServicesSection component snapshot 1`] = `
<div>
  <div
    class="services-section"
  >
    <h2
      class="services-section__header"
    >
      Services I Offer
    </h2>
    <h3
      class="services-section__sub-header"
    >
      A breakdown of all the different kind of services I offer.
    </h3>
    <hr
      class="services-section__line-break"
    />
    <div
      class="services-section__course-container"
    >
      <div
        class="services-card"
      >
        <h1
          class="services-card__title"
        >
          Online Courses
        </h1>
        <h2>
           Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quia quod
 ipsum saepe ipsa optio dolorem veniam tenetur nemo labore ut
        </h2>
        <a
          href="/services"
        >
          <div
            class="services-card__button"
          >
            <img
              alt=""
              class="services-card__vector services-card__vector--black"
              src="Vector-black.svg"
            />
            <img
              alt=""
              class="services-card__vector services-card__vector--white"
              src="Vector-white.svg"
            />
          </div>
        </a>
      </div>
      <div
        class="services-card"
      >
        <h1
          class="services-card__title"
        >
          Group Courses
        </h1>
        <h2>
           Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quia quod
 ipsum saepe ipsa optio dolorem veniam tenetur nemo labore ut
        </h2>
        <a
          href="/services"
        >
          <div
            class="services-card__button"
          >
            <img
              alt=""
              class="services-card__vector services-card__vector--black"
              src="Vector-black.svg"
            />
            <img
              alt=""
              class="services-card__vector services-card__vector--white"
              src="Vector-white.svg"
            />
          </div>
        </a>
      </div>
      <div
        class="services-card"
      >
        <h1
          class="services-card__title"
        >
          Bespoke Packages
        </h1>
        <h2>
           Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quia quod
 ipsum saepe ipsa optio dolorem veniam tenetur nemo labore ut
        </h2>
        <a
          href="/services"
        >
          <div
            class="services-card__button"
          >
            <img
              alt=""
              class="services-card__vector services-card__vector--black"
              src="Vector-black.svg"
            />
            <img
              alt=""
              class="services-card__vector services-card__vector--white"
              src="Vector-white.svg"
            />
          </div>
        </a>
      </div>
    </div>
    <button
      class="button button--secondary"
    >
      VIEW OUR SERVICES
    </button>
  </div>
</div>
`;
